backend/src/Services/SigningService.cs

```diff
@@ -104,31 +123,45 @@ private async Task<string> GetFingerprint()
         return fingerprint;
     }
 
-    private static string WriteDataToFile(string data)
+    private bool WriteDataToFile(string data)
     {
-        string docPath = Path.Combine(Environment.CurrentDirectory, "dataToSign.json");
+        try
+        {
+            using (StreamWriter outputFile = new StreamWriter(FILENAME))
+            {
+                outputFile.WriteLine(data);
+            }
 
-        using (StreamWriter outputFile = new StreamWriter(docPath))
+            return true;
+        }
+        catch (Exception ex)
         {
-            outputFile.WriteLine(data);
+            logger.WriteFileException(ex.ToString());
+            return false;
         }
-
-        return docPath;
     }
 
-    private static string ReadSignatureFromFile(string filepath)
+    private bool ReadSignatureFromFile(out string signature)
     {
-        string signature = "";
-        using (StreamReader file = new StreamReader(filepath + ".asc"))
+        try
+        {
+            using (StreamReader file = new StreamReader(FILENAME + ".asc"))
+            {
+                signature = file.ReadToEnd();
+            }
+            return true;
+        }
+        catch (Exception ex)
         {
-            signature = file.ReadToEnd();
+            logger.ReadFileException(ex.ToString());
+            signature = "";
+            return false;
         }
-        return signature;
     }
 
-    private static void RemoveFiles(string filepath)
+    private static void RemoveFiles()
     {
-        File.Delete(filepath);
-        File.Delete(filepath + ".asc");
+        File.Delete(FILENAME);
+        File.Delete(FILENAME + ".asc");
```

Comment:
Could make the string `".asc"` a constant, for example, `..._FILE_EXTENSION` and make the string `FILENAME + ".asc"` a constant `...FILENAME`.